src/FreeAge/server/unit.cpp

```diff
@@ -14,42 +14,39 @@ ServerUnit::ServerUnit(int playerIndex, UnitType type, const QPointF& mapCoord)
   SetHP(GetUnitMaxHP(type));
 }
 
-void ServerUnit::SetTarget(u32 targetObjectId, ServerObject* targetObject, bool isManualTargeting) {
-  InteractionType interaction = GetInteractionType(this, targetObject);
-  
+void ServerUnit::SetTarget(u32 targetObjectId, ServerObject* targetObject, bool isManualTargeting, InteractionType interaction) {
+  targetObjectInteraction = interaction;
+  if (interaction == InteractionType::Unknown) {
+    interaction = GetInteractionType(this, targetObject);
+    // Keeps the targetObjectInteraction with the value of Unknown.
```

Comment:
I wrote it this way to keep the old behaviour the same but I don't know if the GetInteractionType could return different in the future. If not it could just be stored here don't call it again.